### ISSUE TITLE
Use Cargo to install rust deps into //third_party/rust_crates

### ISSUE THREAD

**Issue by ry** [ry/deno#366]
Currently we are using gclient to mange rust deps:
https://github.com/ry/deno/blob/ad4f335847daffcc5556008aebebaf2d1eecac67/gclient_config.py#L42-L65
These get installed by `tools/build_third_party.py`
https://github.com/ry/deno/blob/ad4f335847daffcc5556008aebebaf2d1eecac67/tools/build_third_party.py#L30
This task would instead call `cargo install` to put that source code into `//third_party/rust_crates`.

I still want to define the build and linking manually in `//build_extra/rust/BUILD.gn`:
https://github.com/ry/deno/blob/master/build_extra/rust/BUILD.gn#L24-L57
This task is only about how to fetch the source code. Deno does not aim to build like a normal rust package at this point. "cargo build" in the root project directory will fail - that's ok.

Part of this task will be creating a `Cargo.toml` file for the project. There are only two third party deps so far: "url" and "libc".

**Comment by robbym** [comment]
I was under the assumption that `cargo install` was strictly for downloading and installing binary packages instead of libraries. Although I get the idea of what you want.

**Comment by robbym** [comment]
`cargo fetch` seems to be the one. Will investigate.

**Comment by robbym** [comment]
This functionality doesn't seem to exist in one place, one tool more specifically.
There is an open issue that addresses it though.
https://github.com/rust-lang/cargo/issues/1861

**Comment by robbym** [comment]
The third-party cargo subcommand `cargo clone` does the right thing: https://github.com/JanLikar/cargo-clone, but its command line arguments only.

**Comment by f-a-a** [comment]
Following from the discussions in #370:
* Manually parsing Cargo.toml and maintaining the program within deno is a no-go (introducing unnecessary complexity)
* Most sensible approach is to introduce a cargo subcommand to achieve:
```sh
cargo download-deps --download_path=third_party/rust_crates/ --config=./Cargo.toml
```
and call it in `./tools/build_third_party.py`. (@robbym created a [POC](https://github.com/robbym/cargo-download-deps))

**Comment by ry** [comment]
Fixed in d93bd4b127c39e2f0e8d2ec3bbec3f7c118890f4
